lib/vrl/compiler/src/value/convert.rs

```diff
@@ -494,3 +494,25 @@ impl From<DateTime<Utc>> for Value {
         Value::Timestamp(v)
     }
 }
+
+impl From<serde_json::Value> for Value {
+    fn from(value: serde_json::Value) -> Self {
+        match value {
+            serde_json::Value::Null => Value::Null,
+            serde_json::Value::Bool(bool) => bool.into(),
+            serde_json::Value::Number(number) => {
+                if number.is_f64() {
```

Comment:
I think we may actually want to follow the same pattern of `event::Value` here:

https://github.com/timberio/vector/blob/206876830474100d6f23a9c498bcf9deccccc104/lib/vector-core/src/event/value.rs#L165-L170

Where it maps to integer first, if it can, then float, then finally string if it can't be represented as an i64 or f64.

Reply:
It does exactly that at the moment, but I agree that it's not obvious from reading the code. From the `is_f64` docs:

> Currently this function returns true if and only if both `is_i64` and
> `is_u64` return false but this is not a guarantee in the future.
>
> ```
> let v = json!({ "a": 256.0, "b": 64, "c": -64 });
>
> assert!(v["a"].is_f64());
>
> // Integers.
> assert!(!v["b"].is_f64());
> assert!(!v["c"].is_f64());
> ```

Reply:
I'm wondering about the string case a bit. It is only ever reached when `n == u64::max_value()`, right?

Reply:
Aha, I missed that note in the docs. It sounds like we shouldn't depend on that behavior though. What you have now looks good to me.

And yeah, I'm not crazy about the string fallback we have for large numbers, but it is the current approach. I think maybe we could swap our internal number formats for a bignum crate or something in the future.